### ISSUE TITLE
How to Make the Universe Think for Us | Quanta Magazine

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#8435]
How to Make the Universe Think for Us | Quanta Magazine<br>
<br>
https://ift.tt/HOTkMpP<br>
<br>
Charlie Wood<br>
<br>
<div>
<div>
<div>
<div>
<div>
<section>
<section>
<div>
<div>
<div>
<h6><a href="https://www.quantamagazine.org/tag/neural-networks/">neural networks</a></h6>
<h1>How to Make the Universe Think for Us</h1>
<div>
<h6>
<em>By</em> <a href="https://www.quantamagazine.org/authors/charlie-wood/"><span>Charlie Wood</span></a>
</h6>
<p><em>May 31, 2022</em></p>
</div>
<div>
<div>Physicists are building neural networks out of vibrations, voltages and lasers, arguing that the future of computing lies in exploiting the universe&rsquo;s complex physical behaviors.</div>
</div>
</div>
</div>
</div>
</section>
</section>
</div>
<figure>
<div>
<div><img alt="An illustration of a spherical universe sitting inside a machine hooked up to wires." src="https://d2r55xnwy6nx47.cloudfront.net/uploads/2022/06/0601_Physical-Neural-Networks_2880x1620_Ledel.jpg"></div>
</div>
<figcaption>
<section>
<div>
<div>
<p>Dan Page for Quanta Magazine</p>
</div>
</div>
</section>
</figcaption>
</figure>
<div>
<div>
<div>
<div>
<section>
<div>
<div>
<div>
<p>Inside a soundproofed crate sits one of the world&rsquo;s worst neural networks. After being presented with an image of the number 6, it pauses for a moment before identifying the digit: zero. <a href="https://www.aep.cornell.edu/faculty-directory/peter-mcmahon">Peter McMahon</a>, the physicist-engineer at Cornell University who led the development of the network, defends it with a sheepish smile, pointing out that the handwritten number looks sloppy. Logan Wright, a postdoc visiting McMahon&rsquo;s lab from NTT Research, assures me that the device usually gets the answer right, but acknowledges that mistakes are common. &ldquo;It&rsquo;s just this bad,&rdquo; he said.</p>
<p>Despite the underwhelming performance, this neural network is a groundbreaker. The researchers tip the crate over, revealing not a computer chip but a microphone angled toward a titanium plate that&rsquo;s bolted to a speaker. Other neural networks operate in the digital world of 0s and 1s, but this device runs on sound. When Wright cues up a new image of a digit, its pixels get converted into audio and a faint chattering fills the lab as the speaker shakes the plate. Metallic reverberations do the &ldquo;reading&rdquo; rather than software running on silicon. That the device often succeeds beggars belief, even to its designers.</p>
<p>&ldquo;Whatever the function of the shaking metal is, it shouldn&rsquo;t have anything to do with classifying a handwritten digit,&rdquo; said McMahon.</p>
<p>The primitive reading ability of the device, which the Cornell group unveiled in <a href="https://www.nature.com/articles/s41586-021-04223-6">a paper in <em>Nature</em> in January</a>, gives McMahon and others hope that its distant descendants could revolutionize computing.</p>
<p>When it comes to conventional machine learning, computer scientists have discovered that <a href="https://www.quantamagazine.org/computer-scientists-prove-why-bigger-neural-networks-do-better-20220210/">bigger is better</a>. Stuffing a neural network with more artificial neurons &mdash; nodes that store numerical values &mdash; improves its ability to tell a dachshund from a Dalmatian, or to succeed at myriad other pattern recognition tasks. Truly tremendous neural networks can pull off unnervingly human undertakings like <a href="https://www.theguardian.com/commentisfree/2020/sep/08/robot-wrote-this-article-gpt-3">composing essays</a> and <a href="https://www.nytimes.com/2022/04/06/technology/openai-images-dall-e.html">creating illustrations</a>. With more computational muscle, even grander feats may become possible. This potential has motivated a multitude of efforts to develop more powerful and efficient methods of computation.</p>
<p>McMahon and a band of like-minded physicists champion an unorthodox approach: Get the universe to crunch the numbers for us. &ldquo;Many physical systems can naturally do some computation way more efficiently or faster than a computer can,&rdquo; McMahon said. He cites wind tunnels: When engineers design a plane, they might digitize the blueprints and spend hours on a supercomputer simulating how air flows around the wings. Or they can stick the vehicle in a wind tunnel and see if it flies. From a computational perspective, the wind tunnel instantly &ldquo;calculates&rdquo; how wings interact with air.</p>
</div>
</div>
</div>
</section>
</div>
</div>
</div>
</div>
<figure>
<div>
<div><img alt="Two men look at a computer screen." src="https://d2r55xnwy6nx47.cloudfront.net/uploads/2022/06/McMahon-and-Onodera.jpg"></div>
</div>
<figcaption>
<section>
<div>
<div>
<p>Peter McMahon and Tatsuhiro Onodera are members of a team at Cornell that has programmed a variety of physical systems to accomplish learning tasks.</p>
</div>
<div>
<p>Dave Burbank for Cornell University</p>
</div>
</div>
</section>
</figcaption>
</figure>
<div>
<section>
<div>
<div>
<div>
<p>A wind tunnel is a single-minded machine; it simulates aerodynamics. Researchers like McMahon are after an apparatus that can learn to do anything &mdash; a system that can adapt its behavior through trial and error to acquire any new ability, such as classifying handwritten digits or distinguishing one spoken vowel from another. Recent work has shown that physical systems like waves of light, networks of superconductors and branching streams of electrons can all learn.</p>
<p>&ldquo;We are reinventing not just the hardware,&rdquo; said <a href="https://mila.quebec/en/person/benjamin-scellier/">Benjamin Scellier</a>, a mathematician at the Swiss Federal Institute of Technology Zurich in Switzerland who helped design a new physical learning algorithm, but &ldquo;also the whole computing paradigm.&rdquo;</p>
<h2><strong>Learning to Think</strong></h2>
<p>Learning is an exotic process; until about a decade ago, brains were the only systems that did it well. It was the structure of the brain that loosely inspired computer scientists to design deep neural networks, now the most popular artificial learning models.</p>
<p>A deep neural network is a computer program that learns through practice. The network can be thought of as a grid: Layers of nodes called neurons, which store values, are connected to neurons in adjacent layers by lines, or &ldquo;synapses.&rdquo; Initially, these synapses are just random numbers known as &ldquo;weights.&rdquo;</p>
<p>When you want the network to read a digit &mdash; 4, say &mdash; you make the first layer of neurons represent a raw image of the 4, perhaps storing the shade of each pixel as a value in a corresponding neuron. Then the network &ldquo;thinks,&rdquo; moving layer by layer, multiplying the neuron values by the synaptic weights to populate the next layer of neurons. The neuron with the highest value in the final layer indicates the network&rsquo;s answer. If it&rsquo;s the second neuron, for instance, the network guesses that it saw a 2.</p>
<p>To teach the network to make smarter guesses, a learning algorithm works backward. After each trial, it calculates the difference between the guess and the correct answer (which, in our example, would be represented by a high value for the fourth neuron in the final layer and low values elsewhere). Then an algorithm steps back through the network layer by layer, calculating how to tweak the weights in order to get the values of the final neurons to rise or fall as needed. This procedure, known as backpropagation, lies at the heart of deep learning.</p>
<p>Through many guess-and-tweak repetitions, backpropagation guides the weights to a configuration of numbers that will, through a cascade of multiplications initiated by an image, spit out the digit written there.</p>
<p>But compared to whatever goes on in the brain, the digitized version of learning that happens in artificial neural networks looks dramatically inefficient. On less than 2,000 calories a day, a human child learns to talk, read, play games, and much more in a few years. On such a restricted energy diet, the groundbreaking GPT-3, a neural network capable of fluent conversation, would have taken a millennium to learn to chat.</p>
<p>From a physicist&rsquo;s perspective, a large digital neural network is simply trying to do too much math. Today&rsquo;s biggest behemoths must record and manipulate more than <a href="https://ai.googleblog.com/2022/04/pathways-language-model-palm-scaling-to.html">half a trillion numbers</a>. The universe, meanwhile, constantly pulls off tasks far beyond the limits of computers&rsquo; meager bookkeeping abilities. A room might have trillions of trillions of air molecules bouncing around; that&rsquo;s an impossible number of moving pieces for a computer to track in a full-fledged simulation of collisions, but the air itself has no trouble deciding how to behave from moment to moment.</p>
<p>The challenge is to build physical systems that can naturally pull off both of the processes needed for AI &mdash; the &ldquo;thinking&rdquo; involved in (say) classifying an image, and the &ldquo;learning&rdquo; needed to classify such images correctly. A system that mastered both tasks would leverage the universe&rsquo;s ability to act mathematically without actually doing math.</p>
<p>&ldquo;We never compute 3.532 times 1.567 or something,&rdquo; Scellier said. &ldquo;It&rsquo;s done, but implicitly, just by the laws of physics directly.&rdquo;</p>
<h2><strong>The Thinking Part</strong></h2>
<p>McMahon and his collaborators have made progress on the &ldquo;thinking&rdquo; half of the puzzle.</p>
<p>While setting up his lab at Cornell in the final months of the pre-pandemic world, McMahon mulled over a curious finding. For years, the top-performing image-recognition neural networks had been getting ever deeper. That is, networks with more layers were better able to take in a bunch of pixels and put out a label, such as &ldquo;poodle.&rdquo; The trend inspired mathematicians to study the transformation (from pixels to &ldquo;poodle&rdquo;) that the networks were achieving, and in 2017 <a href="https://link.springer.com/article/10.1007/s40304-017-0103-z">several</a> <a href="https://arxiv.org/abs/1709.03698">groups</a> <a href="https://arxiv.org/abs/1710.10121">proposed</a> that the networks were acting like approximate versions of a smooth mathematical function. In math, a function turns an input (often a position along the <em>x</em>-axis) into an output (the <em>y</em>-value, or height, of a curve at that position). In a specific type of neural network, more layers do better because the function is less jagged, moving closer to some ideal curve.</p>
<p>The research got McMahon thinking. Perhaps with a smoothly changing physical system, one could sidestep the blockiness inherent in the digital approach.</p>
<p>The trick was to find a way to domesticate a complicated system &mdash; to adapt its behavior with training. McMahon and collaborators chose the titanium plate as one such system because its many patterns of vibrations blend incoming sound in convoluted ways. To make the plate act like a neural network, they fed in one sound that encoded the input image (a handwritten 6, for example), and another representing the synaptic weights; peaks and troughs needed to hit the titanium plate at precisely the right moments for the device to merge the sounds and give the answer &mdash; such as a new sound that&rsquo;s loudest in the sixth millisecond, representing the classification &ldquo;6.&rdquo;</p>
</div>
</div>
</div>
</section>
</div>
<figure>
<div>
<div>
<img alt="A triptych of photos. In the first, a microphone points at a black box containing a small metal plate. In the second, two hands hold a circuit board. In the last, a green laser illuminates mirrors and other optical components." src="https://d2r55xnwy6nx47.cloudfront.net/uploads/2022/06/Three-setups.jpg">
<figcaption>
<section>
<div>
<div>
<p>A team at Cornell University has trained three different physical systems to &ldquo;read&rdquo; handwritten digits. From top, a vibrating titanium plate, a crystal and an electronic circuit.</p>
</div>
<div>
<p>Rob Kurcoba for Cornell University (top and center); Charlie Wood for Quanta Magazine (bottom)</p>
</div>
</div>
</section>
</figcaption>
</div>
</div>
<figcaption>
<section>
<div>
<div>
<p>A team at Cornell University has trained three different physical systems to &ldquo;read&rdquo; handwritten digits: From left, a vibrating titanium plate, a crystal and an electronic circuit.</p>
</div>
<div>
<p>Rob Kurcoba for Cornell University (left and center); Charlie Wood for Quanta Magazine (right)</p>
</div>
</div>
</section>
</figcaption>
</figure>
<div>
<section>
<div>
<div>
<div>
<p>The group also implemented their scheme in an optical system &mdash; where the input image and weights are encoded in two beams of light that get jumbled together by a crystal &mdash; and in an electronic circuit capable of similarly shuffling inputs. In principle, any system with Byzantine behavior will do, though the researchers believe the optical system holds particular promise. Not only can a crystal blend light extremely quickly, but light also contains abundant data about the world. McMahon imagines miniaturized versions of his optical neural network someday serving as the eyes of self-driving cars, identifying stop signs and pedestrians before feeding that information to the vehicle&rsquo;s computer chip, much as our retinas perform some basic visual processing on incoming light.</p>
<p>The Achilles heel of these systems, however, is that training them requires a return to the digital world. Backpropagation involves running a neural network in reverse, but plates and crystals don&rsquo;t readily unmix sounds and light. So the group constructed a digital model of each physical system. Reversing these models on a laptop, they could use the backpropagation algorithm to calculate how to adjust the weights to give accurate answers.</p>
<p>With this training, the plate learned to classify handwritten digits correctly 87% of the time. The circuit and laser reached 93% and 97% accuracy, respectively. The results showed &ldquo;that not only standard neural networks can be trained through backpropagation,&rdquo; said <a href="http://julie.grollier.free.fr/">Julie Grollier</a>, a physicist at the French National Center for Scientific Research (CNRS). &ldquo;That&rsquo;s beautiful.&rdquo;</p>
<p>The group&rsquo;s quivering metal plate has not yet brought computing closer to the shocking efficiency of the brain. It doesn&rsquo;t even approach the speed of digital neural networks. But McMahon views his devices as striking, if modest, proof that you don&rsquo;t need a brain or computer chip to think. &ldquo;Any physical system can be a neural network,&rdquo; he said.</p>
<h2><strong>The Learning Part</strong></h2>
<p>Ideas abound for the other half of the puzzle &mdash; getting a system to learn all by itself.</p>
<p><a href="https://mpl.mpg.de/divisions/marquardt-division/">Florian Marquardt</a>, a physicist at the Max Planck Institute for the Science of Light in Germany, believes one option is to build a machine that runs backward. Last year, he and a collaborator <a href="https://arxiv.org/abs/2103.04992">proposed a physical analogue</a> of the backpropagation algorithm that could run on such a system.</p>
<p>To show that it works, they digitally simulated a laser setup somewhat like McMahon&rsquo;s, with the adjustable weights encoded in a light wave that mixes with another input wave (encoding, say, an image). They nudge the output to be closer to the right answer and use optical components to unmix the waves, reversing the process. &ldquo;The magic,&rdquo; Marquardt said, is that &ldquo;when you try the device once more with the same input, [the output] now has a tendency to be closer to where you want it to be.&rdquo; Next, they are collaborating with experimentalists to build such a system.</p>
<p>But focusing on systems that run in reverse limits the options, so other researchers are leaving backpropagation behind entirely. They take encouragement from knowing that the brain learns in some other way than standard backpropagation. &ldquo;The brain doesn&rsquo;t work like this,&rdquo; said Scellier<strong><em>.</em></strong> Neuron A communicates with neuron B, &ldquo;but it&rsquo;s only one-way.&rdquo;</p>
<p>In 2017, Scellier and <a href="https://yoshuabengio.org/">Yoshua Bengio</a>, a computer scientist at the University of Montreal, developed a unidirectional learning method called <a href="https://www.frontiersin.org/articles/10.3389/fncom.2017.00024/full">equilibrium propagation</a>. To get a sense of how it works, imagine a network of arrows that act like neurons, their direction indicating a 0 or 1, connected in a grid by springs that act as synaptic weights. The looser a spring, the less the linked arrows tend to snap into alignment.</p>
<p>First, you twist arrows in the leftmost row to reflect the pixels of your handwritten digit and hold them fixed while the disturbance ripples out through the springs, flipping other arrows. When the flipping stops, the rightmost arrows give the answer.</p>
<p>Crucially, you don&rsquo;t have to train this system by un-flipping the arrows. Instead, you connect another set of arrows showing the correct answer along the bottom of the network; these flip arrows in the upper set, and the whole grid settles into a new equilibrium. Finally, you compare the new orientations of the arrows with the old orientations and tighten or loosen each spring accordingly. Over many trials, the springs acquire smarter tensions in a way that Scellier and Bengio have shown is equivalent to backpropagation.</p>
<p>&ldquo;It was thought that there was no possible link between physical neural networks and backpropagation,&rdquo; said Grollier. &ldquo;Very recently that&rsquo;s what changed, and that&rsquo;s very exciting.&rdquo;</p>
<p>Initial work on equilibrium propagation was all theoretical. But in an upcoming publication, Grollier and J&eacute;r&eacute;mie Laydevant, a physicist at CNRS, describe an execution of the algorithm on a machine called a quantum annealer, built by the company D-Wave. The apparatus has a network of thousands of interacting superconductors that can act like arrows linked by springs and naturally calculate how the &ldquo;springs&rdquo; should be updated. The system cannot update these synaptic weights automatically, though.</p>
<h2><strong>Closing the Circle</strong></h2>
<p>At least one team has gathered the pieces to build an electronic circuit that does all the heavy lifting &mdash; thinking, learning and updating weights &mdash; with physics. &ldquo;We&rsquo;ve been able to close the loop for a small system,&rdquo; said <a href="https://www.physics.upenn.edu/duriangroup/people/SamDillavou.html">Sam Dillavou</a>, a physicist at the University of Pennsylvania.</p>
<p>The goal for Dillavou and his collaborators is to emulate the brain, a literal smart substance: a relatively uniform system that learns without any single structure calling the shots. &ldquo;Every neuron is doing its own thing,&rdquo; he said.</p>
<p>To this end, they built a self-learning circuit, in which variable resistors act as the synaptic weights and neurons are the voltages measured between the resistors. To classify a given input, it translates the data into voltages that are applied to a few nodes. Electric current courses through the circuit, seeking the paths that dissipate the least energy and changing the voltages as it stabilizes. The answer is the voltage at specified output nodes.</p>
<p>Their major innovation came in the ever-challenging learning step, for which they devised a scheme similar to equilibrium propagation called coupled learning. As one circuit takes in data and &ldquo;thinks up&rdquo; a guess, an identical second circuit starts with the correct answer and incorporates it into its behavior. Finally, electronics connecting each pair of resistors automatically compare their values and adjust them to achieve a &ldquo;smarter&rdquo; configuration.</p>
<p>The group described their rudimentary circuit <a href="https://arxiv.org/abs/2108.00275">in a preprint last summer</a>, showing that it could learn to distinguish three types of flowers with 95% accuracy. Now they&rsquo;re working on a faster, more capable device.</p>
<p>Even that upgrade won&rsquo;t come close to beating a state-of-the-art silicon chip. But the physicists building these systems suspect that digital neural networks &mdash; as mighty as they seem today &mdash; will eventually appear slow and inadequate next to their analog cousins. Digital neural networks can only scale up so much before getting bogged down by excessive computation, but bigger physical networks need not do anything but be themselves.</p>
<p>&ldquo;It&rsquo;s such a big, fast-moving and varied field that I find it hard to believe that there won&rsquo;t be some pretty powerful computers made with these principles,&rdquo; Dillavou said.</p>
</div>
</div>
</div>
</section>
</div>
</div>
</div>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via Quanta Magazine<br>
<br>
June 8, 2022 at 06:00PM
